### PR TITLE
ondeviceorientationabsolute event attribute

### PR DESCRIPTION
**Pull request by plehegar**
https://github.com/w3c/deviceorientation/pull/34

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5149/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/211352984)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 9d9835d1399e9bba17d503ea7f88e02660b8e982
Using browser at version BuildID 20170314110401; SourceStamp 6d38ad302429c98115c354d643e81987ecec5d3c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5149/orientation-event/ondeviceorientationabsolute.html">/orientation-event/ondeviceorientationabsolute.html</a></summary>


|                                  Subtest                                   | Results | Messages |
|----------------------------------------------------------------------------|---------|----------|
|                                                                            | TIMEOUT |          |
| `Provide an event handler IDL attribute named ondeviceorientationabsolute` | NOTRUN  |          |
| `The type of this event handler must be 'DeviceOrientationEvent'`          | NOTRUN  |          |
| `The absolute property must be set to true.`                               | NOTRUN  |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/211352985)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 9d9835d1399e9bba17d503ea7f88e02660b8e982
Using browser at version 58.0.3029.19 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5149/orientation-event/ondeviceorientationabsolute.html">/orientation-event/ondeviceorientationabsolute.html</a></summary>


|                                  Subtest                                   | Results |                      Messages                      |
|----------------------------------------------------------------------------|---------|----------------------------------------------------|
|                                                                            | OK      |                                                    |
| `Provide an event handler IDL attribute named ondeviceorientationabsolute` | PASS    |                                                    |
| `The type of this event handler must be 'DeviceOrientationEvent'`          | FAIL    | `Right-hand side of 'instanceof' is not an object` |
| `The absolute property must be set to true.`                               | FAIL    | `assert_true: expected true got false`             |

</details>
</details>

**Comment by plehegar** [comment]
@ashimura ping...

File: orientation-event/ondeviceorientationabsolute.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      var run = false;
+      window.ondeviceorientationabsolute = function(e) {
+        if (!run) {
+          run = true;
+          t1.step(function() {
+            assert_equals(e.type, "deviceorientationabsolute");
+          });
+          t1.done();
+          t2.step(function() {
+            assert_true(e instanceof "DeviceOrientationEvent");
```
Comment (miketaylr) [inline orientation-event/ondeviceorientationabsolute.html]: s/`"DeviceOrientationEvent"`/`DeviceOrientationEvent`/

File: orientation-event/ondeviceorientationabsolute.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  </head>
+  <body>
+    <p>Rotate the device to run the tests.</p>
+    <div id="log"></div>
+    <script>
+      var t1 = async_test("Provide an event handler IDL attribute named ondeviceorientationabsolute");
+      var t2 = async_test("The type of this event handler must be 'DeviceOrientationEvent'");
+      var t3 = async_test("The absolute property must be set to true.");
+      var run = false;
+      window.ondeviceorientationabsolute = function(e) {
```
Comment (miketaylr) [inline orientation-event/ondeviceorientationabsolute.html]: These tests will timeout if `window.ondeviceorientationabsolute` doesn't exist (and the `deviceorientationabsolute` callback never fires), right?

**Comment by plehegar** [comment]
Thank you @miketaylr .

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20337)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/299113100)

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
@plehegar, can you rebase this to give it another run through Travis?

**Comment by foolip** [comment]
(The lint problem is probably real.)

File: orientation-event/ondeviceorientationabsolute.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>User agents must also provide an event handler IDL attribute named ondeviceorientationabsolute on the window object</title>
+    <meta name=viewport content="width=device-width, maximum-scale=1.0">
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+  </head>
+  <body>
+    <p>Rotate the device to run the tests.</p>
```
Comment (Honry) [inline orientation-event/ondeviceorientationabsolute.html]: Is it a manual test?

**Comment by zcorpan** [comment]
@plehegar ping :)

File: orientation-event/ondeviceorientationabsolute.html (hunk 2 above)
Comment (plehegar) [reply orientation-event/ondeviceorientationabsolute.html]: right. I'll fix that.

File: orientation-event/ondeviceorientationabsolute.html (hunk 1 above)
Comment (plehegar) [reply orientation-event/ondeviceorientationabsolute.html]: oops

File: orientation-event/ondeviceorientationabsolute.html (hunk 3 above)
Comment (plehegar) [reply orientation-event/ondeviceorientationabsolute.html]: Nope. The spec says "If an implementation can never provide orientation information, the event should be fired with the alpha, beta and gamma attributes set to null." (which tested in a separate test).

**Pull request merged**